core/kernel/tee_ta_manager.c

```diff
@@ -416,18 +438,22 @@ TEE_Result tee_ta_close_session(struct tee_ta_session *csess,
 	ctx = sess->ctx;
 	DMSG("Destroy session");
 
+	if (!ctx) {
+		/* The context is already destroyed */
+		tee_ta_destroy_session(sess, open_sessions);
+		return TEE_SUCCESS;
+	} else if (ctx->panicked) {
+		tee_ta_context_destroy(sess->ctx);
+		sess->ctx = NULL;
```

Comment:
I think this should also close the session with `tee_ta_destroy_session(sess, open_sessions)`.

Reply:
In the latest version this has been recoded. Kindly ask you to review the new implementation.